components/eamxx/src/share/field/field_alloc_prop.cpp

```diff
@@ -77,6 +77,56 @@ subview (const int idim, const int k, const bool dynamic) const {
   return props;
 }
 
+FieldAllocProp FieldAllocProp::subview(const int idim,
+                                       const int k_beg,
+                                       const int k_end,
+                                       const bool dynamic) const {
+  EKAT_REQUIRE_MSG(
+      is_committed(),
+      "Error! Subview requires alloc properties to be committed.\n");
+  EKAT_REQUIRE_MSG(idim <= m_layout.rank(),
+                   "Error! Dimension index out of bounds.\n");
+  EKAT_REQUIRE_MSG(k_beg < k_end,
+                   "Error! Slice indices are invalid (non-increasing).\n");
+  EKAT_REQUIRE_MSG(
+      k_beg >= 0 && k_end < m_layout.dim(idim),
+      "Error! Slice index range along the idim dimension is out of bounds.\n");
+
+  // Set new layout basic stuff
+  FieldAllocProp props(m_scalar_type_size);
+  props.m_committed = true;
+  props.m_scalar_type_size = m_scalar_type_size;
+  props.m_layout =
+      m_layout.clone_with_different_extent(idim, k_end - k_beg + 1);
+
+  // Output is contiguous if either
+  //  - this->m_contiguous=true AND idim==0
+  //  - m_layout.dim(i)==1 for all i<idim
+  //  - m_layout.rank()==1 (we end up with a rank-0 subview)
+  auto is_one = [](int i) { return i == 1; };
+  props.m_contiguous = (m_contiguous and idim == 0) || m_layout.rank() == 1 ||
```

Comment:
I'm not sure this is right. As of now, multi-slice subfields will always require strided views. If `m_contiguous` is true, then the `get_view` method of `Field` becomes ok to call, which may yield garbage. I would always consider multi-slice views as non-contiguous, until we figure out a way to be smarter in the `Field::get_view` method(s).